stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/repository/facade/RepositoryFacadeIntegrationTests.kt

```diff
@@ -45,4 +46,40 @@ internal class RepositoryFacadeIntegrationTests : BaseRepositoryFacadeIntegratio
         Truth.assertThat(result).isNotNull()
         Truth.assertThat(result).isEqualTo(message)
     }
+
+    @Test
+    fun `Given a message with theirs reaction When querying message Should return massage without own reactions`() =
+        runBlocking {
+            val messageId = randomString()
+            val user = randomUser(
+                // ignoring fields that are not persisted on purpose
+                totalUnreadCount = 0,
+                unreadChannels = 0,
+                online = false
+            )
+            val theirsReaction = randomReaction(
+                messageId = messageId,
+                type = "haha",
+                user = user,
+                userId = user.id
+            )
+            val message = randomMessage(
+                id = messageId,
+                ownReactions = mutableListOf(),
+                latestReactions = mutableListOf(theirsReaction),
+                reactionCounts = mutableMapOf("haha" to 1),
+                reactionScores = mutableMapOf("haha" to 1),
+            )
+
+            repositoryFacade.insertMessages(listOf(message), cache = false)
+            val result = repositoryFacade.selectMessage(message.id)
+
+            Truth.assertThat(result).isNotNull()
+            Truth.assertThat(result!!.latestReactions).isEqualTo(mutableListOf(theirsReaction))
+            Truth.assertThat(result!!.ownReactions).isEmpty()
```

Comment:
There's no discriminator between `ownReactions` and `latestReaction` when fetching corresponding message relations.